molgenis-data-security/src/main/java/org/molgenis/data/security/meta/EntityTypeRepositorySecurityDecorator.java

```diff
@@ -1,392 +1,54 @@
-
-	@Override
-	public void deleteAll()
-	{
-		iterator().forEachRemaining(entityType ->
-		{
-			this.validateDeleteAllowed(entityType);
-			deleteEntityPermissions(entityType);
-		});
-		super.deleteAll();
-	}
-
-	private void deleteEntityPermissions(EntityType entityType)
-	{
-		deleteEntityPermissions(entityType.getId());
-	}
-
-	private void deleteEntityPermissions(String entityTypeId)
-	{
-		mutableAclService.deleteAcl(new EntityTypeIdentity(entityTypeId), true);
-		mutableAclClassService.deleteAclClass(EntityIdentityUtils.toType(entityTypeId));
-	}
-
-	@Override
-	public void add(EntityType entity)
-	{
-		createAcl(entity);
-		super.add(entity);
-	}
-
-	@Override
-	public Integer add(Stream<EntityType> entities)
-	{
-		return super.add(entities.filter(entityType ->
-		{
-			createAcl(entityType);
-			return true;
-		}));
-	}
-
-	/**
-	 * Updating entityType meta data is allowed for non-system entities. For system entities updating entityType meta data is
-	 * only allowed if the meta data defined in Java differs from the meta data stored in the database (in other words
-	 * the Java code was updated).
-	 *
-	 * @param entityType entity meta data
-	 */
-	private void validateUpdateAllowed(EntityType entityType)
-	{
-		validatePermission(entityType, EntityTypePermission.WRITEMETA);
-
-		boolean isSystem = systemEntityTypeRegistry.hasSystemEntityType(entityType.getId());
-		//FIXME: should only be possible to update system entities during bootstrap!
-		if (isSystem && !currentUserIsSystem())
-		{
-			throw new MolgenisDataException(
-					format("Updating system entity meta data [%s] is not allowed", entityType.getLabel()));
-		}
-	}
-
-	private void validateDeleteAllowed(EntityType entityType)
-	{
-		validatePermission(entityType, EntityTypePermission.WRITEMETA);
-
-		String entityTypeId = entityType.getId();
-		boolean isSystem = systemEntityTypeRegistry.hasSystemEntityType(entityTypeId);
-		if (isSystem)
-		{
-			throw new MolgenisDataException(
-					format("Deleting system entity meta data [%s] is not allowed", entityTypeId));
-		}
-	}
-
-	private void validateDeleteAllowed(Object entityTypeId)
-	{
-		EntityType entityType = findOneById(entityTypeId);
-		if (entityType == null)
-		{
-			throw new UnknownEntityException(
-					format("Unknown entity meta data [%s] with id [%s]", getName(), entityTypeId.toString()));
-		}
-		validateDeleteAllowed(entityType);
-	}
-
-	private void validatePermission(EntityType entityType, EntityTypePermission permission)
-	{
-		boolean hasPermission = permissionService.hasPermission(new EntityTypeIdentity(entityType.getId()), permission);
-		if (!hasPermission)
-		{
-			throw new MolgenisDataAccessException(
-					format("No [%s] permission on entity type [%s] with id [%s]", toMessagePermission(permission),
-							entityType.getLabel(), entityType.getId()));
-		}
-	}
-
-	private static String toMessagePermission(EntityTypePermission permission)
+	protected String toMessagePermission(Operation operation)
```

Comment:
I think we can use toString on the Permission class and let the caller call getPermissionForOperation.

Reply:
Owh and Operation is called Action in our vocabulary now.